### PR TITLE
Fix duplicating/copying `TileMap`

### LINKED ISSUE #88104: `TileMap` can't be duplicated or copied in editor
### Tested versions

- 4.3.dev [41564aaf7708b0bf594f745dd2448a54dd687cc5]

### System information

Godot v4.3.dev (d5481c648) - Windows 10.0.22631 - Vulkan (Forward+) - dedicated AMD Radeon RX 6750 XT (Advanced Micro Devices, Inc.; 31.0.23013.1023) - AMD Ryzen 9 7900X 12-Core Processor (24 Threads)

### Issue description

Attempting to copy or duplicate a `TileMap` with two or more layers fails with (might be in more cases but that's the case I replicated it with, will test with layers with data as well):
```
ERROR: Class 'TileMapLayer' or its base class cannot be instantiated.
   at: (core/object/class_db.cpp:361)
ERROR: Parameter "obj" is null.
   at: Node::_duplicate (scene/main/node.cpp:2523)
ERROR: Condition "!dup" is true. Continuing.
   at: SceneTreeDock::_tool_selected (editor/scene_tree_dock.cpp:755)
```

Will attempt to bisect the issue but it's likely directly related to the restructuring of layers, possibly:
* https://github.com/godotengine/godot/pull/87115

Edit: Can confirm that this is a regression from #87115, will do some further digging later today

### Steps to reproduce

Open the MRP and try to Copy or Duplicate the `TileMap`

### Minimal reproduction project (MRP)

[TileMapDuplicateMRP.zip](https://github.com/godotengine/godot/files/14210276/TileMapDuplicateMRP.zip)

### PR DESCRIPTION
**Pull request by AThousandShips**
Using `force_parent_owned` to prevent issues duplicating

Think this covers all cases

* Fixes: https://github.com/godotengine/godot/issues/88104

### CODE REVIEW

File: scene/register_scene_types.cpp [hunk 1]
```diff
@@ -783,6 +784,7 @@ void register_scene_types() {
 	GDREGISTER_CLASS(TileMapPattern);
 	GDREGISTER_CLASS(TileData);
 	GDREGISTER_CLASS(TileMap);
+	GDREGISTER_INTERNAL_CLASS(TileMapLayer);
```
Comment (AThousandShips) [inline scene/register_scene_types.cpp:787]: Internal as it shouldn't be registered for normal purposes

**Comment by AThousandShips** [comment]
Something with the test project needs tweaking or some exception case needs managing, will look in a bit

**Comment by AThousandShips** [comment]
Appears the internal node is still accessible to scripting and therefore causes some problems, will investigate

**Comment by AThousandShips** [comment]
Will need some tweaking possibly to handle the ability to instance these types, but unsure, the alternative would be to deal somehow with the duplication of the child nodes in a way that prevents breaking, but unsure how that would work best

**Comment by AThousandShips** [comment]
It seems this is relatively trivial, will look at a different fix leveraging `is_owned_by_parent`

**Comment by AThousandShips** [comment]
I've identified a few other places where `force_parent_owned` might be needed and will look at a PR for that too

**Review by Mickeon — APPROVED** [review]
I agree this solution is better. In the future, though, if/when layers are exposed, these snippets should be removed, otherwise you get into a nasty bug. Maybe adding a comment on the side can help?

**Comment by AThousandShips** [comment]
They shouldn't be removed even then, the layers would be doubled because the recreation of the data from the properties

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank *you*!

**Pull request merged**